src/enemy.cpp

```diff
@@ -0,0 +1,118 @@
+#include "enemy.hpp"
+#include <GL/glew.h>
+#include <SDL2/SDL.h>
+#include <array>
+#include <cmath>
+#include <memory>
+#include <string>
+#include <vector>
+#include "camera.hpp"
+#include "linalg.hpp"
+#include "map.hpp"
+#include "opengl_boilerplate.hpp"
+
+static vec4 point_on_surface(float t, float u)
+{
+	using std::cos, std::sin;
+	return vec4{ u*cos(t), 2*(1 - u*u) + u*u*u*0.2f*(1+sin(10*t)), u*sin(t), u };
+}
+
+static const std::vector<std::array<vec4, 3>>& get_vertex_data()
+{
+	static constexpr int tsteps = 150, usteps = 10;
+	static std::vector<std::array<vec4, 3>> vertexdata;
+
+	if (vertexdata.empty()) {
+		float pi = std::acos(-1.0f);
+
+		for (int tstep = 0; tstep < tsteps; tstep++) {
+			for (int ustep = 0; ustep < usteps; ustep++) {
+				// Min value of t tweaked so that the tip of enemy looks good
+				float t1 = lerp<float>(0, 2*pi, tstep/(float)tsteps);
+				float t2 = lerp<float>(0, 2*pi, (tstep+1)/(float)tsteps);
+				float u1 = lerp<float>(0, 1, ustep/(float)usteps);
+				float u2 = lerp<float>(0, 1, (ustep+1)/(float)usteps);
+
+				vec4 a = point_on_surface(t1, u1);
+				vec4 b = point_on_surface(t1, u2);
+				vec4 c = point_on_surface(t2, u1);
+				vec4 d = point_on_surface(t2, u2);
+
+				vertexdata.push_back(std::array<vec4, 3>{a,b,c});
+				vertexdata.push_back(std::array<vec4, 3>{d,b,c});
+			}
+		}
+
+		// Scale it up. No idea why won't work without indexes...
```

Comment:
Maybe it will now work without indexes? I think I was still using `cc` (lol) when I wrote this comment.